presto-main/src/test/java/com/facebook/presto/type/TestTypeRegistry.java

```diff
@@ -169,6 +188,7 @@ public void testGetCommonSuperType()
         assertCommonSuperType("bigint", "decimal(18,0)", "decimal(19,0)");
         assertCommonSuperType("bigint", "decimal(19,0)", "decimal(19,0)");
         assertCommonSuperType("bigint", "decimal(37,1)", "decimal(37,1)");
+        assertCommonSuperType("float", "decimal(37,1)", "float");
```

Comment:
For `double` common type is `double`. Direct, simpler coercion is preferred. So `decimal` coerces to `float` instead of `float`->`double` and `decimal`->`double` coercion.
I know in some cases it would be better to have a coercion to a wider type but then we would end up with much more complicated coercion system.
What's more each coercion is a potential lose of precision, so the fewer the better.